test/integration/test_integration.py

```diff
@@ -11,54 +14,69 @@
 Result = namedtuple('Result', ['outcome', 'artifact', 'item'])
 
 
-def parse_results(log):
-    '''
-    From the given stdout log, parse the results
-    '''
-    start = 'results:'
-    results = []
-    record = False
+class MockEnv:
+    '''Use this to work with mock. Mutliple instances are not safe.'''
+
+    def __init__(self):
+        self.mock = ['mock', '-r', './mock.cfg']
+        self._run(['--init'], check=True)
+
+    def _run(self, what, **kwargs):
+        return subprocess.run(self.mock + what, **kwargs)
+
+    def copy_in(self, files):
+        self._run(['--copyin'] + files + ['/'], check=True)
+
+    def copy_out(self, directory, *, clean_target=False):
+        if clean_target:
+            with contextlib.suppress(FileNotFoundError):
+                shutil.rmtree(directory)
+        self._run(['--copyout', directory, directory], check=True)
+
+    def shell(self, command):
+        self._run(['--shell', command])
+
+    def orphanskill(self):
+        self._run(['--orphanskill'])
 
-    for line in log.splitlines():
-        if line.strip() == start:
-            record = True
-        if record:
-            results.append(line)
-            if not line:
-                break
 
-    if not results:
-        raise RuntimeError('Could not parse output')
-    return yaml.load('\n'.join(results))['results']
+@pytest.fixture(scope="session")
+def mock():
+    '''Setup a mock we can run Ansible tasks in under root'''
+    mockenv = MockEnv()
+    files = ['taskotron_python_versions'] + glob.glob('*.py') + ['tests.yml']
+    mockenv.copy_in(files)
+    yield mockenv
+    mockenv.orphanskill()
```

Comment:
I'm not sure if this is the right way of doing this, but I found not way to "kill all processes from mock".

Note that sometimes I had problem when I tried to <kbd>Ctrl</kbd>+<kbd>C</kbd> the tests, that it only killed the mock process and the entire pytest thing detached from my stdin, so it was running in background and I could not <kbd>Ctrl</kbd>+<kbd>C</kbd> it any more. This is very inconvenient, so I recommend running he integration tests with `--maxfail=1` when running interactively.